.github/workflows/build.yml

```diff
@@ -50,42 +48,34 @@ jobs:
         poetry run scripts/alembic_freeze.py
 
     - name: Release Marker
-      if: ${{ github.event.release }}
+      if: startsWith(github.ref, 'refs/tags/')
       # The release marker differentiates installations 'in the wild' versus internal dev builds and tests
       run: touch src/meltano/core/tracking/.release_marker
 
     - name: Build distribution
-      run: |
-        poetry build --format sdist
-        poetry run pip wheel --no-deps . --wheel-dir dist/
-
-    - name: Upload artifacts
-      uses: actions/upload-artifact@v4
-      with:
-        name: python-artifacts
-        path: 'dist/*'
+      uses: hynek/build-and-inspect-python-package@v2
+      id: baipp
 
   pypi_release:
     name: Publish to PyPI
+    if: startsWith(github.ref, 'refs/tags/')
```

Comment:
I find this a bit concerning, since there's a difference between pushing a git tag (which is not protected) versus creating & publishing a release. Maybe it's still worth triggering this workflow for

```
release:
    types: [published]
```

and then changing behaviour based on what triggered the workflow.